modules/spring-config/src/main/java/org/mule/runtime/config/spring/MuleArtifactContext.java

```diff
@@ -447,6 +488,30 @@ protected OptionalObjectsController getOptionalObjectsController() {
     return currentMuleContext;
   }
 
+  public void initializeComponent(String componentName) {
+    if (muleContext.getRegistry().get(componentName) != null) {
+      return;
+    }
+    MinimalApplicationModelGenerator minimalApplicationModelGenerator =
+        new MinimalApplicationModelGenerator(this.applicationModel, componentBuildingDefinitionRegistry);
+    ApplicationModel minimalApplicationModel;
+    if (!componentName.contains("/")) {
+      minimalApplicationModel = minimalApplicationModelGenerator.getMinimalModelByName(componentName);
+    } else {
+      minimalApplicationModel = minimalApplicationModelGenerator.getMinimalModelByPath(componentName);
+    }
+    createComponents((DefaultListableBeanFactory) this.getBeanFactory(), minimalApplicationModel, false);
+  }
+
+  public ConnectivityTestingService getConnectivityTestingService() {
+    ConnectivityTestingService connectivityTestingService =
+        muleContext.getRegistry().lookupObject(OBJECT_CONNECTIVITY_TESTING_SERVICE);
+    if (enableLazyInit) {
```

Comment:
Why is not the LazyConnectivityTestingService configured in the muleContext instead of having to create a new one here?